### PR TITLE
Close #6550: html: Allow to use HTML permalink texts

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature
- Refactoring

### Purpose
- Add new configuration variables: `html_permalinks` and `html_permalinks_icon`.
- This refines the settings around HTML permalinks.
  * html_add_permalinks
    * Deprecated.
  * html_permalinks
    * Enable or disable permalinks feature.
  * html_permalinks_icon
    * Change the icon for permalinks
- refs: #6550

### CODE REVIEW

File: sphinx/builders/html/__init__.py [hunk 1]
```diff
@@ -1189,6 +1189,16 @@ def validate_html_favicon(app: Sphinx, config: Config) -> None:
         config.html_favicon = None  # type: ignore
 
 
+def migrate_html_add_permalinks(app: Sphinx, config: Config) -> None:
```
Comment (tk0miya) [inline sphinx/builders/html/__init__.py:1208]: I will add a warning for migration on 4.0 release (no warnings during 3.x).

File: sphinx/writers/html.py [hunk 2]
```diff
@@ -842,3 +838,9 @@ def depart_math_block(self, node: Element, math_env: str = '') -> None:
 
     def unknown_visit(self, node: Node) -> None:
         raise NotImplementedError('Unknown node: ' + node.__class__.__name__)
+
+    @property
+    def permalink_text(self) -> str:
```
Comment (tk0miya) [inline sphinx/writers/html.py:839]: I have to mention this depracation in CHANGES.

**Review by shimizukawa — APPROVED** [review]
LGTM. 

My previous view is here: https://github.com/sphinx-doc/sphinx/pull/6550#issuecomment-751435197
Based on the previous discussion, I have no objection this time.

File: doc/extdev/deprecated.rst [hunk 3]
```diff
@@ -71,7 +71,17 @@ The following is a list of deprecated interfaces.
...
+     - 3.5
+     - 5.0
+     - :confval:`html_permalinks_icon`
+
+   * - ``sphinx.writers.html5.HTML5Translator.permalink_text``
+     - 3.5
+     - 5.0
+     - :confval:`html_permalinks_icon`
+
+* - The ``follow_wrapped`` argument of ``sphinx.util.inspect.signature()``
```
Comment (tk0miya) [inline doc/extdev/deprecated.rst]: Oops... this is merging miss...

**Comment by tk0miya** [comment]
Merged. @shimizukawa Thank you for reviewing!

**Pull request merged**